packages/block-editor/src/components/preview-options/plugin-preview-menu-item/index.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies
+ */
+import { Fill, MenuItem } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { check } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { coreDeviceTypes } from '../index';
+
+/**
+ * Component used by a plugin to define the contents of a menu item that
+ * selects a "custom preview". The children of this component will be displayed
+ * inside the preview menu. Typically a single string is good enough.
+ *
+ * @param {Object} props          Component properties.
+ * @param {string} props.previewId The internal name of this custom preview. Must match the _previewId_ given to `PluginPreview`.
+ * @param {WPElement} props.children Children to be rendered.
+ */
+export default function PluginPreviewMenuItem( {
+	children,
+	previewId,
+	...props
+} ) {
+	const {
+		__experimentalSetPreviewDeviceType: setPreviewDeviceType,
+	} = useDispatch( 'core/edit-post' );
+
+	const { deviceType } = useSelect(
+		( select ) => ( {
+			deviceType: select(
+				'core/edit-post'
```

Comment:
The block editor is a generic package that may be used outside WordPress, while the core/edit-post is a package to edit regular WordPress posts. I think we should avoid having the block-editor depending or querying data from core/edit-post.

I guess things will also not work as expected for edit-site scree where we don't have a core/edit-post module?


The device type is passed to the PreviewOptions as a prop. Could we rely on that prop instead?


Not directly related to this PR, but I think it makes sense to move the preview functionality under the WordPress/interface package. It is common to edit-post and edit-site. It has functionality like the singleEnableItems that can be used to store which preview is active. I guess the slot / fill here could also be abstracted as action item packages/interface/src/components/action-item/README.md.

Regarding the API, we are exposing PluginPreview, PluginPreviewMenuItem on the @wordpress/block-editor' package. As far as I know, the bock-editor does not expose slot & fills besides the ones used by blocks. All the plugin related API's are exposed under edit-post/edit-site. I guess for this case it may make sense too to make PluginPreview, PluginPreviewMenuItem under @wordpress/edit-site/edit-post. I guess we could implement the slot fills under WordPress/interface, and then on edit-site and edit-post, we would simply have wrappers that pass a specific scope. Similar to what we do for the sidebar and menu item slots.
The PreviewOptions component would have a prop for additional preview options that edit-site and edit-post use.

cc: @gziolo and @youknowriad In case you have a different option.